src/frontend/src/test-e2e/views.ts

```diff
@@ -606,6 +608,19 @@ export class DemoAppView extends View {
     await this.browser.$("#signinBtn").click();
   }
 
+  /** Waits for the authentication to finish, the window to close and the principal to update.
+   * Returns the principal after the user has been authenticated.
+   */
+  async waitForAuthenticated(): Promise<string> {
+    // wait for the demo app to close the II window
+    await waitToClose(this.browser);
```

Comment:
this will break if we ever have more than one window open (could picture this happening with VCs); not saying this should be changed but good to keep in mind